server/app/views/v1/grid_services/_grid_service.json.jbuilder

```diff
@@ -33,9 +33,9 @@ json.stack do
   json.id grid_service.stack.to_path
   json.name grid_service.stack.name
 end
-json.links grid_service.grid_service_links.map{|s|
-  { id: s.linked_grid_service.to_path, alias: s.alias, name: s.linked_grid_service.name }
-}
+json.links grid_service.grid_service_links.map { |s|
+  { id: s.linked_grid_service.to_path, alias: s.alias, name: s.linked_grid_service.name } if s.linked_grid_service
```

Comment:
I don't understand what you mean... the `GridServices::Delete` mutation should still only be the only thing that destroys any `GridService` model, and that validates that the service is not linked to anymore? Apart from the bug of the `Stacks::Update` -> `StackDeployWorker` skipping the `GridService` validations... which still needs to be fixed...